tests/components/azure_event_hub/test_init.py

```diff
@@ -0,0 +1,211 @@
+"""The tests for the Azure Event Hub component."""
+from collections import namedtuple
+
+import pytest
+
+import homeassistant.components.azure_event_hub as azure_event_hub
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+
+from tests.async_mock import MagicMock, patch
+
+AZURE_EVENT_HUB_PATH = "homeassistant.components.azure_event_hub"
+PRODUCER_PATH = f"{AZURE_EVENT_HUB_PATH}.EventHubProducerClient"
+MIN_CONFIG = {
+    "event_hub_namespace": "namespace",
+    "event_hub_instance_name": "name",
+    "event_hub_sas_policy": "policy",
+    "event_hub_sas_key": "key",
+}
+FilterTest = namedtuple("FilterTest", "id should_pass")
+
+
+@pytest.fixture(autouse=True, name="mock_client", scope="module")
+def mock_client_fixture():
+    """Mock the azure event hub producer client."""
+    with patch(f"{PRODUCER_PATH}.send_batch") as mock_send_batch, patch(
+        f"{PRODUCER_PATH}.close"
+    ) as mock_close, patch(f"{PRODUCER_PATH}.__init__", return_value=None) as mock_init:
+        yield (
+            mock_init,
+            mock_send_batch,
+            mock_close,
+        )
+
+
+@pytest.fixture(autouse=True, name="mock_batch")
+def mock_batch_fixture():
+    """Mock batch creator and return mocked batch object."""
+    mock_batch = MagicMock()
+    with patch(f"{PRODUCER_PATH}.create_batch", return_value=mock_batch):
+        yield mock_batch
+
+
+@pytest.fixture(autouse=True, name="mock_policy")
+def mock_policy_fixture():
+    """Mock azure shared key credential."""
+    with patch(f"{AZURE_EVENT_HUB_PATH}.EventHubSharedKeyCredential") as policy:
+        yield policy
+
+
+@pytest.fixture(autouse=True, name="mock_event_data")
+def mock_event_data_fixture():
+    """Mock the azure event data component."""
+    with patch(f"{AZURE_EVENT_HUB_PATH}.EventData") as event_data:
+        yield event_data
+
+
+@pytest.fixture(autouse=True, name="mock_call_later")
+def mock_call_later_fixture():
+    """Mock async_call_later to allow queue processing on demand."""
+    with patch(f"{AZURE_EVENT_HUB_PATH}.async_call_later") as mock_call_later:
+        yield mock_call_later
+
+
+async def test_minimal_config(hass):
+    """Test the minimal config and defaults of component."""
+    config = {azure_event_hub.DOMAIN: MIN_CONFIG}
+    assert await async_setup_component(hass, azure_event_hub.DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)